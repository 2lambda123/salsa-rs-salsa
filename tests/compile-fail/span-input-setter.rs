#[salsa::input]
pub struct MyInput {
    field: u32,
}

fn main() {
    let mut db = salsa::default_database();
    let input = MyInput::new(&mut db, 22);
    input.field(&db);
    input.set_field(22);
}
