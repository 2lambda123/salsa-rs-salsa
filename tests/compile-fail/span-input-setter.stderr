error[E0308]: mismatched types
  --> tests/compile-fail/span-input-setter.rs:26:21
   |
26 |     input.set_field(22);
   |           --------- ^^ expected `&mut dyn Db`, found integer
   |           |
   |           arguments to this method are incorrect
   |
   = note: expected mutable reference `&mut dyn Db`
                           found type `{integer}`
note: method defined here
  --> tests/compile-fail/span-input-setter.rs:18:5
   |
16 | #[salsa::input]
   | ---------------
17 | pub struct MyInput {
18 |     field: u32,
   |     ^^^^^
